roles/openshift_hosted/tasks/storage/glusterfs.yml

```diff
@@ -25,6 +25,18 @@
   set_fact:
     openshift_hosted_registry_fsgroup: "{{ registry_pods.results.results[0]['items'][0].spec.securityContext.fsGroup }}"
 
+- name: Get registry PersistentVolume
+  oc_obj:
+    namespace: "{{ openshift_hosted_registry_namespace }}"
+    state: list
+    kind: pv
+    name: "{{ openshift_hosted_registry_storage_volume_name }}-glusterfs-volume"
+  register: registry_pv
+
+- name: Determine registry volume path
+  set_fact:
+    openshift_hosted_registry_storage_glusterfs_path: "{{ openshift_hosted_registry_storage_glusterfs_path | default(registry_pv.results.results[0].spec.glusterfs.path) }}"
```

Comment:
openshift_hosted_registry_storage_glusterfs_path is always defined, so this condition will never be true.  It is currently defined in roles/openshift_facts/defaults/main.yml

Reply:
Remove this set fact.

On mount task in question "Mount registry volume" below...

Add a vars section to bottom of task like this:
```
vars:
  l_registry_storage_glusterfs_path: "{{ l_registry_storage_glusterfs_path_discover.results.results[0].spec.glusterfs.path | default(openshift_hosted_registry_storage_glusterfs_path) }}"
```

Change openshift_hosted_registry_storage_glusterfs_path variable in the task to the one we created in vars.